### PR TITLE
chore: parameterize Synth

### PR DESCRIPTION
**Pull request by ssddOnTop**
same as: #2435 
but it uses `jit::Result<Value>`

Fixes: #2463
/claim #2463

### CODE REVIEW

File: src/core/jit/synth/synth.rs [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
-    selection: Vec<Field<Nested<ConstValue>, ConstValue>>,
-    store: Store<Result<ConstValue>>,
-    variables: Variables<ConstValue>,
+use crate::core::jit::model::{ExecutionPlan, Field, Nested, Variable, Variables};
+use crate::core::jit::store::{Data, DataPath, Store};
+use crate::core::jit::synth::Synthesizer;
+use crate::core::json::{JsonLikeOwned, JsonObjectLike};
+
+// TODO: rename
+pub struct AlsoSynth<Value: JsonLikeOwned> {
```
Comment (tusharmath) [inline src/core/jit/synth/synth.rs:8]:
```suggestion
pub struct AlsoSynth<Value> {
```

Add bounds to impl always and not to constructors.

File: src/core/jit/synth/synth.rs [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
+pub struct Synth<Value> {
+    selection: Vec<Field<Nested<Value>, Value>>,
+    store: Store<Result<Value>>,
+    variables: Variables<async_graphql_value::ConstValue>,
 }
 
 impl<Extensions, Input> Field<Extensions, Input> {
     #[inline(always)]
-    pub fn skip(&self, variables: &Variables<ConstValue>) -> bool {
+    pub fn skip(&self, variables: &Variables<async_graphql_value::ConstValue>) -> bool {
```
Comment (tusharmath) [inline src/core/jit/synth/synth.rs:40]: Hard dependency on ConstValue should be dropped from here.

**Comment by ssddOnTop** [comment]
waiting for https://github.com/tailcallhq/tailcall/pull/2442 to get merged

File: src/core/jit/synth/synth.rs [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    plan: OperationPlan<Value>,
+}
+
+impl<Value: JsonLikeOwned> AlsoSynth<Value> {
+    pub fn new(plan: OperationPlan<Value>) -> Self {
+        Self { plan }
+    }
+}
+
+impl<Value: JsonLikeOwned + Clone> Synthesizer for AlsoSynth<Value> {
```
Comment (tusharmath) [inline src/core/jit/synth/synth.rs:21]: Now that Synth has been parameterized, we don't need the `Synthesizer` trait anymore.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**